### ISSUE TITLE
Bad issue with rerouting IPs

### ISSUE THREAD

**Issue by Gray1989** [ekristen/dns-proxy#3]
You need to make sure that domain.indexOf(s) does not equal -1 in your server.js file in order to fix the problem. The more IPs that are being routed, the more prominent the issue is. Also, include your github repo in with your json file so that users can get here easily from the npm site.

**Comment by ekristen** [comment]
Can you share your configuration with me.

I'd also like to understand the behavior you are seeing a bit more too. I have several entries in my domains config and I don't see a problem.

Thanks!

**Comment by Gray1989** [comment]
Sure, no problem. The configuration below also causes `mail.google.com` (as-well as many others) to result in the IP address I had set for `www.facebook.com` (192.168.0.17):

    {"port":53,"host":"192.168.0.17","logging":"dnsproxy:query","nameservers":["208.67.222.222","208.67.220.220"],"servers":{},"domains":{"www.facebook.com":"192.168.0.17"},"hosts":{},"fallback_timeout":350}

This is because since the string `mail.google.com` does not contain the string `www.facebook.com`, the result of `domain.indexOf(s)` is `-1`... This is checked against the calculation of the length of the string `mail.google.com` (which is 15) subtracted by the length of the string `www.facebook.com` (which is 16), which also equals `-1`. Since the result of the calculation matches the index, it is assumed to be a match despite the fact that there is no relation whatsoever aside from the length of the strings.

Basically any preset domain settings are mistakenly matching any domains with exactly one less character. But thanks for the quick response, great job with the project otherwise!

**Comment by ekristen** [comment]
@Gray1989 thanks for the info, I understand your problem now. I can definitely fix that.

**Comment by Gray1989** [comment]
No problem. Just thought I would point out an issue I already fixed on my end. I actually ended up making quite a few non-breaking changes to the code in order to get it working the way I want it. I would be willing to become a contributor to the project, I just have little experience actually using git-hub. If you are interested though let me know.

**Comment by ekristen** [comment]
I'm open to changes. Fork the repo and commit your changes against your fork and then do a PR against mine. I'll review, or you can do multiple PRs.
